src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php

```diff
@@ -1227,6 +1252,71 @@ public function findServiceGroupsByHostAndService(int $hostId, int $serviceId):
         return $serviceGroups;
     }
 
+    /**
+     * Find downtimes for host or service
+     * @param int $hostId
+     * @param int $serviceId
+     * @return array
+     */
+    public function findDowntimes(int $hostId, int $serviceId)
+    {
+        $downtimes = [];
+
+        if (empty($hostId)) {
+            return $downtimes;
+        }
+
+        $sql = 'SELECT * FROM `:dbstg`.`downtimes` WHERE host_id = :hostId AND service_id = :serviceId ' .
+                'AND deletion_time IS NULL';
```

Comment:
you have to set some limitation of the result here

Reply:
ok thanks, applied